tf2onnx/tfonnx.py

```diff
@@ -1720,6 +1720,39 @@ def mknode(op_type, inputs, opname, **kwargs):
     return nodes
 
 
+def sparse_softmax_cross_entropy_with_logits_op(ctx, node, name, args):
+    nodes = []
+
+    def mknode(op_type, inputs, outputs=None, **kwargs):
+        op_name = utils.make_name(op_type)
+        if outputs is None:
+            outputs = [port_name(op_name)]
+        new_node = Node(helper.make_node(op_type, inputs, outputs, op_name, **kwargs), ctx)
+        nodes.append(new_node)
+        return new_node
+
+    # make subgraph to implement one_hot, idea comes from onehot_op
+    indices_name = node.input[1]
+    indices_shape = ctx.get_shape(indices_name)
+    if len(indices_shape) != 1:
+        # TODO: this works for rank=1 but tensorflow supports more than this.
+        # Same principle should work but we need to implemtn our own eye.
+        raise ValueError("onehot op: only rank1 is supported")
+    depth = node.inputs[0].output_shapes[0][-1]
```

Comment:
use ctx.get_shape() please for coding style consistency.

Reply:
resolved